spec/unit/base/Version.spec.js

```diff
@@ -151,3 +153,220 @@ describe("done should only be called once in each", () => {
     expect(mockDone.mock.calls.length).toBe(1);
   });
 });
+
+describe("each method", function () {
+  let holodeck;
+  let client;
+  const bodyOne = {
+    next_page_uri: "/2010-04-01/Accounts/AC123/Messages.json?Page=1",
+    messages: [{ body: "payload0" }, { body: "payload1" }],
+  };
+  const bodyTwo = {
+    next_page_uri: "/2010-04-01/Accounts/AC123/Messages.json?Page=2",
+    messages: [{ body: "payload2" }, { body: "payload3" }],
+  };
+  const bodyThree = {
+    next_page_uri: null,
+    messages: [{ body: "payload4" }],
+  };
+
+  beforeEach(function () {
+    holodeck = new Holodeck();
+    client = new Twilio("ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX", "AUTHTOKEN", {
+      httpClient: holodeck,
+    });
+  });
+
+  it("should call user callback foreach resource instance", function (done) {
+    let mockCallback = jest.fn();
+    holodeck.mock(new Response(200, bodyOne));
+    holodeck.mock(new Response(200, bodyTwo));
+    holodeck.mock(new Response(200, bodyThree));
```

Comment:
Could `holodeck.mock` be inside `beforeEach`, since its repeating in everything testcase

Reply:
Good question. Although most of the unit tests have the same mocked responses, not all of them do and there is no guarantee that future tests in this test group that covers the `each` function may need the same mocked responses. Therefore, I think it's best to limit the setup of each test (in the `beforeEach`) to only variables and states that are needed for all tests.

Reply:
Sounds good to me, thanks for clarifying!